Documentation/How to Prepare a Publish Profile.md

```diff
@@ -0,0 +1,64 @@
+# How to Prepare a Publish Profile
+
+In order to deploy applications to a development board (e.g. Raspberry Pi, HummingBoard, etc.), you need to setup a Publish Profile.  Below provides a few steps to quickly get started wihin your projects.
+
+1. Once you have an application setup in Visual Studio, right-click the project in the Solution Explorer and select Publish...
+
+2. In the Pick a publish target dialog, select Folder, choose a folder to publish your application files and click Create Profile.  The example below shows a path of C:\PublishedApps.
+
+3. A default profile, FolderProfile.pubxml, will now be created and added to your project.  You can view it in the Solution Explorer under your project > Properties > PublishProfiles folder.
+
+4. It is a good practice to rename your profile to something you can relate with your project.  In the Publish Window, click the Actions dropdown and select Rename Profile.  A Rename Profile dialog prompts you to rename your profile.  Click Save after renaming.
+
+5. You can configure the profile's settings by clicking Configure... in the Publish Window.  A Profile Settings dialog prompt includes a few options.  
+  
+    **Notes**  
+    * Currently, the Target Runtime doesn't offer a selection for Linux ARM.  You will need to manually open the profile's XML and change the RuntimeIdentifier element to **linux-arm** shown below:
+
+        ```
+        <RuntimeIdentifier>linux-arm</RuntimeIdentifier>
+        ```  
+  
+    * Deployment Mode Options:
+        * **Framework Dependent** - App relies on the presence of a shared system-wide version of .NET Core on the target system.  This will create a smaller size package.
+        * **Self-contained** - .NET Core and required libraries will be bundled with your application creating a larger size package.
```

Comment:
I think it might also be convenient if we add some instructions on how to trim down the self-contained app. Usually, disk space is a concern when dealing with these sort of devices, and having a trimmed down self contained app might reduce as much as 85% of the app size (about ~50 mb)

Reply:
I agree.  I'll be honest and not sure on content or even how to.  I read somewhere there was going to be some better tooling for this in .NET Core 3 time frame.  I also thought I came across a blog, but the tooling was more around Mono for now.  If you know the steps and some content to read over, send it my way and I'll try to figure how to add to this doc.  Thx

Reply:
There are no solutions on how to do this natively with the tooling yet, but you can add an extra package reference that will take care of this for you. In order to trim your selfcontained app, all you need is to add the following to your .csproj:

```xml
  <PropertyGroup>
    <TrimUnusedDependencies>true</TrimUnusedDependencies>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Packaging.Tools.Trimming" Version="1.1.0-preview1-26619-01" />
  </ItemGroup>
```